File: src/main/java/com/adrabazha/air/attack/alarm/system/dto/DistrictStateObject.java
package com.adrabazha.air.attack.alarm.system.dto;

import com.adrabazha.air.attack.alarm.system.model.AlarmState;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class DistrictStateObject {

    private String districtName;

    private String districtCode;

    private AlarmState alarmState;
}

File: src/main/frontend/src/dto/DistrictStateObject.js
import AlarmState from "./AlarmState.ts";

class DistrictStateObject {

    districtName: string
    districtCode: string
    alarmState: AlarmState

    constructor(record: any) {
        this.districtName = record.districtName;
        this.districtCode = record.districtCode;
        this.alarmState = AlarmState[record.alarmState];
    }
}

export default DistrictStateObject